source/common/common/utility.h

```diff
@@ -110,6 +111,22 @@ class RealTimeSource : public TimeSource {
   MonotonicTime monotonicTime() override { return std::chrono::steady_clock::now(); }
 };
 
+/**
+ * Class used for creating non-memory allocating std::ostream.
+ */
+class FixedSizeStreamBuffer : public std::streambuf {
```

Comment:
I see that this is used to initialize a std::streambuf before std::ostream; can FixedSizeStreamBuffer wrap instead of inherit?